pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -70,21 +78,37 @@ def get_profiled_train_dataloader(self, train_dataloader) -> Iterable:
     def prepare_data(self) -> None:
         # on multi-gpu jobs we only want to manipulate (download, etc) on node_rank=0, local_rank=0
         # or in the case where each node needs to do its own manipulation in which case just local_rank=0
-        if self.can_prepare_data():
-            if self.trainer.datamodule is not None:
-                self.trainer.datamodule.prepare_data()
+        should_call_once_per_node = self.trainer.local_rank == 0
+        should_call_once_in_total = self.trainer.local_rank == 0 and self.trainer.node_rank == 0
+
+        # handle datamodule prepare data:
+        # check for prepare_data_per_node & datamodule lifecycle properties before calling datamodule.prepare_data
+        if self.trainer.datamodule is not None:
+            if is_overridden("prepare_data", self.trainer.datamodule) and (
```

Comment:
tbh i don't think we need this `is_overridden` check at all since the default implementation of `prepare_data` is a no-op. i'm not sure why it was added before